enterprise/management/commands/assign_enterprise_user_roles.py

```diff
@@ -0,0 +1,145 @@
+
+    Example usage:
+        $ ./manage.py assign_enterprise_user_roles --role=enterprise_admin
+        $ ./manage.py assign_enterprise_user_roles --role=enterprise_learner
+    """
+    help = 'Assigns enterprise roles to existing enterprise users.'
+
+    def add_arguments(self, parser):
+        """
+        Entry point for subclassed commands to add custom arguments.
+        """
+        parser.add_argument(
+            '--role',
+            action='store',
+            dest='role',
+            default=None,
+            help='Role to assign users role assignments.'
+        )
+        parser.add_argument(
+            '--batch-limit',
+            action='store',
+            dest='batch_limit',
+            default=100,
+            help='Number of users in each batch of conditional offer migration.',
+            type=int,
+        )
+
+        parser.add_argument(
+            '--batch-offset',
+            action='store',
+            dest='batch_offset',
+            default=0,
+            help='Which index to start batching from.',
+            type=int,
+        )
+
+        parser.add_argument(
+            '--batch-sleep',
+            action='store',
+            dest='batch_sleep',
+            default=10,
+            help='How long to sleep between batches.',
+            type=int,
+        )
+
+    def _get_enterprise_admin_users_batch(self, start, end):
+        """
+        Returns a batched queryset of User objects.
+        """
+        logger.info('Fetching new batch of enterprise admin users from indexes: %s to %s', start, end)
+        return User.objects.filter(groups__name=ENTERPRISE_DATA_API_ACCESS_GROUP, is_staff=False)[start:end]
+
+    def _get_enterprise_customer_users_batch(self, start, end):
+        """
+        Returns a batched queryset of EnterpriseCustomerUser objects.
+        """
+        logger.info('Fetching new batch of enterprise customer users from indexes: %s to %s', start, end)
+        return EnterpriseCustomerUser.objects.all()[start:end]
+
+    def handle(self, *args, **options):
+        """
+        Entry point for managment command execution.
+        """
+        enterprise_admin_role = SystemWideEnterpriseRole.objects.get(name=ENTERPRISE_ADMIN_ROLE)
+        enterprise_learner_role = SystemWideEnterpriseRole.objects.get(name=ENTERPRISE_LEARNER_ROLE)
+
+        role = options['role']
+        batch_limit = options['batch_limit']
+        batch_sleep = options['batch_sleep']
+        batch_offset = options['batch_offset']
+
+        current_batch_index = batch_offset
+        logger.info('Starting assigning enterprise roles to users!')
+
+        # Assign admin role to non-staff users with enterprise data api access.
+        if role == ENTERPRISE_ADMIN_ROLE:
+            enterprise_admin_users = self._get_enterprise_admin_users_batch(
+                batch_offset,
+                batch_offset + batch_limit
+            )
+            while len(enterprise_admin_users) > 0:
```

Comment:
seems like you could DRY this up and consolidate with some of the logic below

Reply:
@brittneyexline addressed :)